# Zadanie domowe
**Cel**: Stworzenie notatnika Jupyter, w którym krok po kroku wykonasz analizę temperatur z użyciem podstawowych struktur i instrukcji Pythona.

Dla zadeklarowanych danych
1. Używając pętli `for`, wypisz temperatury.

2. Używając pętli `for`, wypisuj temperatury do momentu pierwszej wyższej niż 25°C (włącznie).

3. Dla każdej temperatury wykonaj sprawdzenie:
- Jeśli temperatura jest niższa niż 20°C, wypisz „chłodno”.
- Jeśli temperatura jest w przedziale 20–25°C, wypisz „przyjemnie”.
- Jeśli temperatura jest wyższa niż 25°C, wypisz „gorąco”.

4. Napisz funkcję `srednia_temperatura`, która przyjmie listę temperatur i zwróci ich średnią wartość. Uruchom funkcję dla podanej listy temperatur i wypisz wynik (do policzenia średniej użyj funkcji `sum` - sumuje liczby, funkcja `len` zwraca długość listy).

5. Napisz funkcję `dni_temperatura_powyzej`, która przyjmie słownik temperatury_dni oraz próg temperatury i zwróci listę dni, w których temperatura była wyższa niż podany próg. Uruchom funkcję dla progu 22°C i wypisz wynik.

6. Używając list comprehension, utwórz nową listę temperatur zwiększonych o 2°C i wypisz wynik.

7. Używając list comprehension, utwórz listę temperatur wyższych niż 20°C i wypisz wynik.

8. W osobnej komórce markdown podsumuj wyniki:

- Średnia temperatura.
- Najcieplejszy dzień i jego temperatura.
- Lista dni z temperaturą powyżej 22°C.

In [2]:
# Lista temperatur z ostatniego tygodnia
temperatury = [15, 20, 22, 19, 30, 25, 18]

# Słownik - przypisanie temperatur do dni tygodnia
temperatury_dni = {
    "Pon": 15,
    "Wt": 20,
    "Śr": 22,
    "Czw": 19,
    "Pt": 30,
    "Sob": 25,
    "Nd": 18
}

In [3]:
# 1.
for temperatura in temperatury:
    print(temperatura)

15
20
22
19
30
25
18


In [4]:
# 2.
for temperatura in temperatury:
    print(temperatura)
    if temperatura > 25:
        break

15
20
22
19
30


In [7]:
#3. Tu nie trzeba było wypisywać kontretnie temperatury
for temperatura in temperatury:
    if temperatura < 20:
        print(temperatura, "chłodno")
    elif temperatura <= 25:
        print(temperatura, "przyjemnie")
    elif temperatura > 25:
        print(temperatura, "gorąco")

15 chłodno
20 przyjemnie
22 przyjemnie
19 chłodno
30 gorąco
25 przyjemnie
18 chłodno


In [10]:
#4.
def srednia_temperatura(lista):
    return sum(lista) / len(lista)

print(srednia_temperatura(temperatury))

21.285714285714285


In [11]:
#5.
def dni_temperatura_powyzej(slownik, prog):
    dni_powyzej = []
    for key, value in slownik.items():
        if value > prog:
            dni_powyzej += [key]
    return dni_powyzej

dni_temperatura_powyzej(temperatury_dni, 22)

['Pt', 'Sob']

In [12]:
#6.
nowa_lista = [x + 2 for x in temperatury]
print(nowa_lista)

[17, 22, 24, 21, 32, 27, 20]


In [13]:
#7.
nowa_lista_20 = [x for x in temperatury if x > 20]
print(nowa_lista_20)

[22, 30, 25]


# 8.
Średnia temperatura : `21.285714285714285`
Najcieplejszy dzień i jego temperatura : `Pt`, `30`
Lista dni z temperaturą powyżej 22°C. : `Pt`, `Sob`